src/System.Private.Interop/src/InteropExtensions/NativeCallableAttribute.cs

```diff
@@ -7,7 +7,7 @@ namespace System.Runtime.InteropServices
 {
     //BARTOK expects
     [AttributeUsage(AttributeTargets.Method)]
-    public sealed class NativeCallableAttribute : Attribute
+    internal sealed class NativeCallableAttribute : Attribute
```

Comment:
The better change is to remove this file from System.Private.Interop project when compiling coreclr.

Reply:
We don't need this type at all.  This type has been defined in coreclr framework.

Reply:
Removing it from System.Private.Interop is not an option as code withing System.Private.Interop needs this class.

If i keep this class public then foll: is the error we get below error.
Compiling generated source code: /usr/share/dotnet/sdk/2.0.0/Roslyn/RunCsc.sh /noconfig @"/home/naveen/temp/mcgOUT\Intermediates\/ConsoleApp.McgInterop.rsp"
mcgOUT\Intermediates\/ConsoleApp.McgInterop/ImplTypes.g.cs(167,44): error CS0433: The type 'NativeCallableAttribute' exists in both 'System.Private.Interop, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' and 'System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'
mcgOUT\Intermediates\/ConsoleApp.McgInterop/ImplTypes.g.cs(197,44): error CS0433: The type 'NativeCallableAttribute' exists in both 'System.Private.Interop, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' and 'System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'

Reply:
sorry for confusion. what I means, remove the type definition in System.Private.Interop.dll. The CoreCLR System.Private.Interop.dll doesn't need this type, since its framework library already contains this type.